hamocc/inventory_bgc.F90

```diff
@@ -350,32 +353,45 @@ SUBROUTINE INVENTORY_BGC(kpie,kpje,kpke,dlxp,dlyp,ddpo,omask,iogrp)
 !     & (zocetratot(idet)+zocetratot(idoc)+zocetratot(iphy)               &
 !     & +zocetratot(izoo))*rcar+zocetratot(isco212)+zocetratot(icalc)
 
-  totalcarbon=                                                                 &
-       & (zocetratot(idet)+zocetratot(idoc)+zocetratot(iphy)                   &
-       & +zocetratot(izoo))*rcar+zocetratot(isco212)+zocetratot(icalc)         &
-       & +zpowtratot(ipowaic)+zsedlayto(isssc12)+zsedlayto(issso12)*rcar       &
-       & +zburial(isssc12)+zburial(issso12)*rcar                               &
-       & +zprorca*rcar+zprcaca                                                 &
-#if defined(BOXATM)
-       & +zatmco2*ppm2con
-#else
-       & +co2flux
-#endif
-
-  totalnitr=                                                                   &
-       &   (zocetratot(idet)+zocetratot(idoc)+zocetratot(iphy)                 &
-       &  +zocetratot(izoo))*rnit+zocetratot(iano3)+zocetratot(igasnit)*2      &
-       &  +zpowtratot(ipowno3)+zpowtratot(ipown2)*2                            &
-       &  +zsedlayto(issso12)*rnit+zburial(issso12)*rnit                       &
-       &  +zocetratot(ian2o)*2                                                 &
-       &  - sndepflux                                                          &
-       &  +zprorca*rnit                                                        &
-#if defined(BOXATM)
-       &  +zatmn2*ppm2con*2
-#else
-
-       &  +sn2flux*2+sn2oflux*2
-#endif
+  if (use_BOXATM) then
```

Comment:
Here and also for the other species: Instead of repeating all contributions, I would suggest to write once the general contributions and then, afterward:
```
if (use_boxatm) then
        totalcarbon = totalcarbon + zatmco2*ppm2con
else
        totalcarbon = totalcarbon + co2flux
endif
```
which is easier to maintain, shorter and clearer, what the difference is.

Reply:
Great idea! I'll implement that.